### PR TITLE
[OpenVINO-EP] UEP v3.1 Release with OpenVINO 2021.4

### PR DESCRIPTION
**Pull request by MaajidKhan**
**Description**: 
This is OpenVINO EP V3.1 Release with OpenVINO 2021.4 version

**Motivation and Context**
- some simple bug fixes
- code clean up (Removed duplicate code at places)
- Removed support for OpenVINO 2020.3 LTS and OpenVINO 2021.1 version
- Enabled CI logs (Required these logs for internal Unit Testing). This are only enabled when the environment variable is 
  set to 1
- Fixed Yolo object Detection Models. As part of this effort, most of the Yolo models are now fully supported on OpenVINO-EP 
backend.
- Minor Dockerfile fix added for ubuntu dockerfile

### CODE REVIEW

**Comment by jywu-msft** [comment]
the description mentions 2021.4 ,can you clarify if it is an updated version of 2021.4 (i.e. 2021.4.1 )
or this is just monthly UEP fixes that are for the same version of OpenVINO 2021.4 that https://github.com/microsoft/onnxruntime/pull/8369 added support for

File: onnxruntime/test/providers/cpu/nn/conv_transpose_op_test.cc [hunk 1]
```diff
@@ -255,7 +255,8 @@ TEST(ConvTransposeTest, ConvTranspose_2D_OutputShape_1) {
                         18.0f, 27.0f, 27.0f, 18.0f,
                         18.0f, 27.0f, 27.0f, 18.0f,
                         12.0f, 18.0f, 18.0f, 12.0f};
-  TestConvTransposeOp(attrs, {X, W}, {X_shape, W_shape}, expected_vals, Y_shape);
+  TestConvTransposeOp(attrs, {X, W}, {X_shape, W_shape}, expected_vals, Y_shape,
```
Comment (jywu-msft) [inline onnxruntime/test/providers/cpu/nn/conv_transpose_op_test.cc:258]: see: https://github.com/microsoft/onnxruntime/blob/d12874086d6c8f6ee6a4070bcfd094b2caea7bd0/onnxruntime/test/providers/cpu/nn/conv_transpose_op_test.cc#L33
unfortunately, the default seems to exclude TensorRTExecutionProvider which this overrides. 
can you add both OpenVINOExecutionProvider and TensorRTExecutionProvider for exclusion?
This is causing TensorRT CI pipeline to fail.

File: onnxruntime/test/providers/cpu/nn/conv_transpose_op_test.cc [hunk 2]
```diff
@@ -30,7 +30,7 @@ void TestConvTransposeOpInitializer(const ConvTransposeOpAttributes& attributes,
                                     bool is_filter_initializer = false,
                                     OpTester::ExpectResult expect_result = OpTester::ExpectResult::kExpectSuccess,
                                     const std::string& err_str = "",
-                                    const std::unordered_set<std::string>& excluded_provider_types = {kTensorrtExecutionProvider}) {
+                                    const std::unordered_set<std::string>& excluded_provider_types = {kTensorrtExecutionProvider, kOpenVINOExecutionProvider}) {
```
Comment (jywu-msft) [inline onnxruntime/test/providers/cpu/nn/conv_transpose_op_test.cc:33]: but you've overrided this in all the TestConvTransposeOp() calls below?
you should undo all your modifications there.

Comment (MaajidKhan) [reply onnxruntime/test/providers/cpu/nn/conv_transpose_op_test.cc:33]: Got it George. I will revert the last commit.
I see, there are 2 failing convtranspose test with TensorRT, I will disable these 2 specific tests for TensorRT.

2021-08-30T19:46:08.6933821Z 1: [  FAILED  ] 2 tests, listed below:
2021-08-30T19:46:08.6934377Z 1: [  FAILED  ] **ConvTransposeTest.ConvTranspose_2D_OutputShape_1**
2021-08-30T19:46:08.6934994Z 1: [  FAILED  ] **ConvTransposeTest.ConvTranspose_2D_OutputShape_1_group_2_for_tranpose_path**

**Review by jywu-msft — APPROVED** [review]

**Pull request merged**